.grit/patterns/no_self_assigned_variables.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Variables should not be self-assigned
+---
+
+# Variables should not be self-assigned
+
+It is redundant and usually a bug when a variable is assigned to itself.
+
+tags: #java
+
+```grit
+language java
+
+assignment_expression($left, $right) as $assignment where {
```

Comment:
We want to generally prefer snippets where possible. Does `$left = $right` not work?